cpp/src/io/csv/reader_impl.cu

```diff
@@ -349,7 +349,12 @@ table_with_metadata reader::impl::read(size_t range_offset,
       if (column_types[active_col].id() == type_id::EMPTY) {
         column_types[active_col] = data_type{STRING};
       }
-      out_buffers.emplace_back(column_types[active_col], num_records, true, stream, mr_);
+      out_buffers.emplace_back(
+        column_types[active_col],
+        num_records,
+        true,
+        stream,
+        column_types[active_col].id() != type_id::STRING ? mr_ : rmm::mr::get_default_resource());
```

Comment:
please add a comment to explain why this is done. Alternatively, you can add a boolean variable whose name would help understand the logic here. Something like:
`auto const is_final_allocation = column_types[active_col].id() != type_id::STRING;`

Reply:
I pushed the change you suggested.